cli/azd/pkg/config/manager.go

```diff
@@ -27,9 +28,75 @@ func NewManager() Manager {
 	return &manager{}
 }
 
+const infraParametersKey = "infra.parameters"
+const vaultIdKey = "vaultId"
+
 // Saves the azd configuration to the specified file path
 func (c *manager) Save(config Config, writer io.Writer) error {
-	configJson, err := json.MarshalIndent(config.Raw(), "", "  ")
+	_, hasInfraParams := config.Get(infraParametersKey)
+	secretKeys := config.SecretKeys()
+	if !hasInfraParams || len(secretKeys) == 0 {
+		// no infra parameters, or no secrets in config
+		configJson, err := json.MarshalIndent(config.Raw(), "", "  ")
+		if err != nil {
+			return fmt.Errorf("failed marshalling config JSON: %w", err)
+		}
+		_, err = writer.Write(configJson)
+		return err
+	}
+
```

Comment:
It feels like we shouldn't have to care about if there are infra parameters or not.  We should only have to ask if the Config object holds any secrets.

Reply:
I wonder if we should be doing something like updating the signature of Save to instead of taking an io.Writer taking some new abstraction that understands how to write two types of values, plaintext values and secret values, and the push all this vault logic behind that interface...